File: java/string-2/xyzThere.java
/* Return true if the given string contains an appearance of "xyz" where the 
 * xyz is not directly preceeded by a period (.). So "xxyz" counts but 
 * "x.xyz" does not.
 */
public boolean xyzThere(String str) {
    if(str.length() >= 3 && str.substring(0, 3).equals("xyz"))
        return true;
          
    for(int i = 1; i < str.length() - 2; i++) {
        if(str.charAt(i - 1) != '.' && str.substring(i, i + 3).equals("xyz"))
            return true;
    }
                          
    return false;
}

File: python/string-2/xyz_there.py
# Return True if the given string contains an appearance of "xyz" where the 
# xyz is not directly preceeded by a period (.). So "xxyz" counts but "x.xyz" 
# does not.
def xyz_there(str):
    if str[:3] == "xyz":
        return True
                    
    for i in range(1, len(str) - 2):
        if str[i-1] != "." and str[i:i+3] == "xyz":
            return True
                                      
    return False
